core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,96 @@
   }
+}
 
-  def isDebugEnabled: Boolean = logger.isDebugEnabled
+trait Logging extends LazyLogging {
+  def loggerName: String = this.getClass.getName
 
-  def isTraceEnabled: Boolean = logger.isTraceEnabled
+  protected var logIdent: String = _
 
-  def debug(msg: => String): Unit = {
-    if (logger.isDebugEnabled())
-      logger.debug(msgWithLogIdent(msg))
-  }
-  def debug(e: => Throwable): Any = {
-    if (logger.isDebugEnabled())
-      logger.debug(logIdent,e)
-  }
-  def debug(msg: => String, e: => Throwable) = {
-    if (logger.isDebugEnabled())
-      logger.debug(msgWithLogIdent(msg),e)
-  }
-  def swallowDebug(action: => Unit) {
-    CoreUtils.swallow(logger.debug, action)
-  }
+  Log4jControllerRegistration
 
-  def info(msg: => String): Unit = {
-    if (logger.isInfoEnabled())
-      logger.info(msgWithLogIdent(msg))
-  }
-  def info(e: => Throwable): Any = {
-    if (logger.isInfoEnabled())
-      logger.info(logIdent,e)
-  }
-  def info(msg: => String,e: => Throwable) = {
-    if (logger.isInfoEnabled())
-      logger.info(msgWithLogIdent(msg),e)
-  }
-  def swallowInfo(action: => Unit) {
-    CoreUtils.swallow(logger.info, action)
-  }
+  protected def msgWithLogIdent(msg: String): String =
+    if (logIdent == null) msg else logIdent + msg
 
-  def warn(msg: => String): Unit = {
-    logger.warn(msgWithLogIdent(msg))
-  }
-  def warn(e: => Throwable): Any = {
-    logger.warn(logIdent,e)
-  }
-  def warn(msg: => String, e: => Throwable) = {
-    logger.warn(msgWithLogIdent(msg),e)
-  }
-  def swallowWarn(action: => Unit) {
-    CoreUtils.swallow(logger.warn, action)
-  }
-  def swallow(action: => Unit) = swallowWarn(action)
+  def trace(msg: => String): Unit = logger.trace(msgWithLogIdent(msg))
 
-  def error(msg: => String): Unit = {
-    logger.error(msgWithLogIdent(msg))
-  }		
-  def error(e: => Throwable): Any = {
-    logger.error(logIdent,e)
-  }
-  def error(msg: => String, e: => Throwable) = {
-    logger.error(msgWithLogIdent(msg),e)
-  }
-  def swallowError(action: => Unit) {
-    CoreUtils.swallow(logger.error, action)
-  }
+  // The return type here should be Unit as in any other logging function in this class.
+  // However making it so will cause the compiler to believe that the signature of this
+  // and the trace above will be the same as the parameters are generic Function objects
+  // subject to type erasure. After the erasure the signature of the two function would
+  // be the same.
+  def trace(e: => Throwable): Any = logger.trace(logIdent,e)
 
-  def fatal(msg: => String): Unit = {
-    logger.fatal(msgWithLogIdent(msg))
-  }
-  def fatal(e: => Throwable): Any = {
-    logger.fatal(logIdent,e)
-  }	
-  def fatal(msg: => String, e: => Throwable) = {
-    logger.fatal(msgWithLogIdent(msg),e)
-  }
+  def trace(msg: => String, e: => Throwable): Unit = logger.trace(msgWithLogIdent(msg),e)
+
+  def swallowTrace(action: => Unit): Unit = CoreUtils.swallow(logger.underlying.trace, action)
```

Comment:
This seems unused. We should remove any unused method that is unusual as that will make it easier for us to remove this class.

Reply:
Ok, agreed

Reply:
Turns out it is used in `RequestSendThread.doWork` around line 213, although through CoreUtils (how neat :) ). So I I think we should leave it for KAFKA-6187